src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs

```diff
@@ -185,13 +186,26 @@ public Task<bool> LaunchTestHostAsync(TestProcessStartInfo testHostStartInfo, Ca
 
         EqtTrace.Verbose("DefaultTestHostmanager: Full path of {0} is {1}", testHostProcessName, testhostProcessPath);
 
+        var processName = _processHelper.GetCurrentProcessFileName()!;
         var launcherPath = testhostProcessPath;
         if (!_environment.OperatingSystem.Equals(PlatformOperatingSystem.Windows) &&
-            !_processHelper.GetCurrentProcessFileName()!.EndsWith(DotnetHostHelper.MONOEXENAME, StringComparison.OrdinalIgnoreCase))
+            !processName.EndsWith(DotnetHostHelper.MONOEXENAME, StringComparison.OrdinalIgnoreCase))
         {
             launcherPath = _dotnetHostHelper.GetMonoPath();
             argumentsString = testhostProcessPath.AddDoubleQuote() + " " + argumentsString;
         }
+        else
+        {
+            // Patching the relative path for IDE scenarios.
+            if (_environment.OperatingSystem.Equals(PlatformOperatingSystem.Windows)
+                && !(processName.EndsWith("dotnet", StringComparison.OrdinalIgnoreCase)
```

Comment:
I used `EndsWith` instead of equals in order to be consistent with this check here: https://github.com/microsoft/vstest/blob/b2e2126f1aa7e5753cafe9515563c99ade6a59ce/src/Microsoft.TestPlatform.CrossPlatEngine/TestEngine.cs#L610

I'm not sure if it was written like that to handle some corner case but I find this approach more flexible than `Equals` and I think `Equals` brings little benefit overall in this case.